transactions.go

```diff
@@ -111,10 +111,88 @@ func (t ItemTransaction) Mutate(b *Battle) {
 		}
 		// TODO: remove consumed item from party's inventory
 	}
+	switch t.Item.Category() {
+	case ItemCategoryHealing, ItemCategoryRevival, ItemCategoryStatusCures:
+		if t.Item.Category() == ItemCategoryRevival && target.CurrentHP != 0 {
+			return
+		}
+		fIndex := target.Friendship / 100
+		data, ok := medicineData[t.Item]
+		if ok {
+			if data.Heal > 0 {
+				b.QueueTransaction(HealTransaction{
+					Target: target,
+					Amount: data.Heal,
+				})
+			}
+			if amount := data.Friendship[fIndex]; amount != 0 {
+				b.QueueTransaction(FriendshipTransaction{
+					Target: target,
+					Amount: amount,
+				})
+			}
+			if data.Cure != StatusNone {
+				b.QueueTransaction(CureStatusTransaction{
+					Target:       t.Target,
+					StatusEffect: data.Cure,
+				})
+			}
+		}
+		// Other medicine item effects
+		switch t.Item {
+		case ItemSacredAsh:
+			for _, pkmn := range b.parties[t.Target.party].pokemon() {
+				if pkmn.CurrentHP == 0 {
+					b.QueueTransaction(HealTransaction{
+						Target: pkmn,
+						Amount: pkmn.MaxHP(),
+					})
+				}
+			}
+		case ItemFullRestore, ItemMaxPotion, ItemMaxRevive, ItemRevivalHerb:
+			b.QueueTransaction(HealTransaction{
+				Target: target,
+				Amount: target.MaxHP(),
+			})
+		case ItemRevive:
+			b.QueueTransaction(HealTransaction{
+				Target: target,
+				Amount: target.MaxHP(),
+			})
```

Comment:
If you use `^uint(0)`, you can put these in `medicineData`

Reply:
I think this is an inaccuracy because why wouldn't we just track the actual value instead? Also, I need the max HP anyway for revive and this actually pointed out that my test was wrong for revival items 👀